app/pages/project/vpcs/VpcsPage.tsx

```diff
@@ -54,6 +55,17 @@ export function VpcsPage() {
   const { data: vpcs } = usePrefetchedApiQuery('vpcList', { query: { project, limit: 25 } })
   const navigate = useNavigate()
 
+  const colHelper = createColumnHelper<Vpc>()
+  const staticCols = [
+    colHelper.accessor('name', { cell: makeLinkCell((vpc) => pb.vpc({ project, vpc })) }),
+    colHelper.accessor('dnsName', { header: 'DNS name' }),
+    colHelper.accessor('description', {}),
+    colHelper.accessor('timeCreated', {
+      header: 'created',
+      cell: (info) => <DateCell value={info.getValue()} />,
+    }),
+  ]
+
```

Comment:
This one, on the other hand, has a column that can't be defined outside of render because it relies on `project`. In this case, we need to memoize.

```ts
const columns = useMemo(() => [
  colHelper.accessor('name', { cell: makeLinkCell((vpc) => pb.vpc({ project, vpc }))}),
  colHelper.accessor('dnsName', { header: 'DNS name' }),
  colHelper.accessor('description', {}),
  colHelper.accessor('timeCreated', {
    header: 'created',
    cell: (info) => <DateCell value={info.getValue()} />,
  }),
  getActionsCol(makeActions)
], [project, makeActions])
```

There's no point using `useColsWithActions` here — all it really does is wrap that `useMemo`. We may find we're using `useColsWithActions` in so few places after this change that we should just get rid of it.

In theory we could move the four statically definable columns out of here and spread them in with `...staticCols`, but I don't think the misdirection is worth it. Because of the `useMemo` they are only ever defined once anyway.